### ISSUE TITLE
`$ref` support, with relative paths for `$id`.

### ISSUE THREAD

**Issue by danywu** [coveooss/json-schema-for-humans#155]
I have a bunch of schemas that use relative paths for `$id`s, e.g. `/components/0.1/crs`, being referenced in `/elements/0.1/points`, and so on. There's a folder structure that reflects those paths in the `$id`s. These schemas will be "deployed" to various environments, e.g. dev, staging, etc. I would like to keep the `$id`s relative for flexibility.

In the documentation, I could not find a way of specifying the base URI for all the schemas in the "top level". Is this something that is supported, or planned to be?

**Comment by pspot2** [comment]
@danywu  are you already using this feature for validation, e.g. are you already using a validator where you can set the base URI externally and you would like to re-use the same schema files both for validation and for documentation generation?

If not, then the only way would be to set it statically in the `$id` of the top-level schema to something universal that would be present in all envs (for example, `/var/tmp/something`). This should make any JSON schema software happy, including this tool.

**Comment by danywu** [comment]
> @danywu are you already using this feature for validation, e.g. are you already using a validator where you can set the base URI externally and you would like to re-use the same schema files both for validation and for documentation generation?

Yes, we are using this for validation already, by setting the base URI. 

> If not, then the only way would be to set it statically in the `$id` of the top-level schema to something universal that would be present in all envs (for example, `/var/tmp/something`). This should make any JSON schema software happy, including this tool.

Our schemas will be served through CDN, so the path will depend on the environments, e.g. ourschemas.com, dev.ourschemas.com, etc.
